windows-toolkit/WindowsCommunityToolkit
# Created SmokeTest projects.

## Fixes #3448

This new SmokeTest project builds once for each value of the ToolkitPackages variable in SmokeTests.proj.
It automatically adds the local built nuget packages and adds the MainPage.[ToolkitPackage].xaml.cs and MainPage.[ToolkitPackage].xaml file, for each project built.

With this, we can analyze the size of the appx bundle that is built for each smoke test and compare that with other built versions.

To test this:
- Run the build.bat script to make sure you have updated nuget files locally
- Open a VS cmd or pwsh and from the SmokeTests folder, run "msbuild .\SmokeTests.proj".
  - It will take a while to build it all, since it is set to build for Release(.Net Native) and it will generate the appxbundle files on the SmokeTests/AppPackages folder, one for each project.

## PR Type
What kind of change does this PR introduce?



- Feature 


- Build or CI related changes 





## What is the current behavior?
No way to analyze the assembly size of a simple project that reference each nuget package.

## What is the new behavior?
The build pipeline automatically builds a simple project for each 'ToolkitPackages', allowing us to analyze the size impact of each PR, or from time to time.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [X] Tested code with current [supported SDKs](../readme.md#supported)
- [ ] Pull Request has been submitted to the documentation repository [instructions](..\contributing.md#docs). Link: 
- [ ] Sample in sample app has been added / updated (for bug fixes / features)
    - [ ] Icon has been created (if new sample) following the [Thumbnail Style Guide and templates](https://github.com/windows-toolkit/WindowsCommunityToolkit-design-assets)
- [ ] Tests for the changes have been added (for bug fixes / features) (if applicable)
- [X] Header has been added to all new source files (run *build/UpdateHeaders.bat*)
